plugins/modules/ip_rdns.py

```diff
@@ -82,28 +82,53 @@ def __init__(self) -> None:
             required_if=[["state", "present", ["rdns"]]],
         )
 
-    def update_rdns(self, rdns: str) -> None:
+    def _should_update_rdns(
+        self, old_rdns: str, new_rdns: Union[str, ExplicitNullValue]
+    ) -> bool:
+        """
+        Returns whether the old RDNS value and the proposed new RDNS for the
+        IP address differ.
+        """
+
+        # If the RDNS address is being reset, compare the old RDNS against
+        # the Linode API default
+        if isinstance(new_rdns, ExplicitNullValue):
+            ip_address = self.module.params.get("address")
+            new_rdns = (
+                f"{ip_address.replace('.', '-')}.ip.linodeusercontent.com"
```

Comment:
I'm not the biggest fan of hardcoding the default RDNS address but I'm not aware of a better way to work around it as of now

Reply:
It seems the way we have to work around the default RDNS. Another thing I can think of is makeing this logic into a helper function like `build_default_rdns(...)`, may help a bit with readability.

Reply:
I like the idea of a helper function, I'll push up a commit to fix that 👍